Assets/Scripts/OnCollision.cs

```diff
@@ -0,0 +1,14 @@
+using UnityEngine;
+public class OnCollision : MonoBehaviour
+{
+    void OnCollisionEnter2D(Collision2D collision)
+    {
+        if (collision.gameObject.CompareTag("Enemy"))
+        {
+            gameObject.GetComponent<Player>().health -= 1;
+            Destroy(collision.gameObject);
+            Debug.Log("Player has been hit, health is now: " + gameObject.GetComponent<Player>().health);
+        }
+    }
+
+}
```

Comment:
Oops, thought I was done with my review, but there are actually more changes I'd like you to make haha.

Instead of using `gameObject.GetComponent<Player>().health`, we can turn it into a variable as it'll be easier to read and one require the `GetComponent` function to run every time we need to refer to it. To do this, let's make a `Player player` variable that will hold our player object.

The `Start()` function in `MonoBehaviour` is basically like an _init_ function. This is where you want to initialize your `player` variable by using `GetComponent` here. We can then refer to the player health from now on as simply `player.health` instead of `gameObject.GetComponent<Player>().health`

So like this:

```
using UnityEngine;

public class OnCollision : MonoBehaviour
{
    private Player player;

    private void Start()
    {
        player = gameObject.GetComponent<Player>();
    }

    private void OnCollisionEnter2D(Collision2D collision)
    {
        if (collision.gameObject.CompareTag("Enemy"))
        {
            player.health -= 1;
            Destroy(collision.gameObject);
            Debug.Log("Player has been hit, health is now: " + player.health);

            if (player.health <= 0)
                Debug.Log("Game Over");
        }
    }
}
```

Reply:
Also don't forget to make your functions private unless otherwise needed by other classes!

Reply:
Done